apps/server/src/shared/infra/identity-management/keycloak/uc/Keycloak-management.uc.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from '@nestjs/common';
+import fs from 'node:fs/promises';
+import { IJsonAccount, IJsonUser } from '../interface';
+import { KeycloakAdministrationService } from '../keycloak-administration.service';
+
+@Injectable()
+export class KeycloakManagementUc {
+	constructor(private readonly kcAdmin: KeycloakAdministrationService) {}
+
+	public async check(): Promise<boolean> {
+		return this.kcAdmin.testKcConnection();
+	}
+
+	public async clean(): Promise<number> {
+		let kc = await this.kcAdmin.callKcAdminClient();
+		const users = (await kc.users.find()).filter(
+			(user) => user.id !== undefined && user.username !== this.kcAdmin.getAdminUser()
+		);
+
+		// eslint-disable-next-line no-restricted-syntax
+		for (const user of users) {
+			// eslint-disable-next-line no-await-in-loop
+			kc = await this.kcAdmin.callKcAdminClient();
+			// eslint-disable-next-line no-await-in-loop
+			await kc.users.del({
+				// can not be undefined, see filter above
+				id: user.id ?? '',
+			});
+		}
+		return users.length;
+	}
+
+	async seed(): Promise<number> {
+		let userCount = 0;
+		const users = await this.loadUsers();
+		const accounts = await this.loadAccounts();
+		// eslint-disable-next-line no-restricted-syntax
+		for (const user of users) {
+			const account = accounts.find((a) => a.userId.$oid === user._id.$oid);
+			if (account) {
+				// eslint-disable-next-line no-await-in-loop
+				const kc = await this.kcAdmin.callKcAdminClient();
+				// eslint-disable-next-line no-await-in-loop
+				await kc.users.create({
```

Comment:
Possible optimization with `Promise.all` possible? But it's only for seeding, right? So about 60-70 users. So it's a highly optional optimization.

Reply:
Yeah its for seeding purpose only. Also, these steps must be executed sequentially: Ideally, we would encapsulate the entire 'kc' object and it's signature. Again, that comes with some excessive boilerplate coding. Since the module already hides all details of 'kc', I figured that acceptable.